Content.Shared/Turnstile/Systems/TurnstileSystem.cs

```diff
@@ -0,0 +1,180 @@
+﻿using Content.Shared.Mobs.Components;
+using Content.Shared.Physics;
+using Content.Shared.Pulling.Components;
+using Content.Shared.Turnstile.Components;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Physics.Events;
+using Robust.Shared.Physics.Systems;
+
+namespace Content.Shared.Turnstile.Systems;
+
+public sealed class TurnstileSystem : EntitySystem
+{
+    [Dependency] private readonly SharedPhysicsSystem _physicsSystem = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearanceSystem = default!;
+    [Dependency] private readonly EntityLookupSystem _entityLookupSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<TurnstileComponent, ComponentInit>(OnComponentInit);
+        SubscribeLocalEvent<TurnstileComponent, StartCollideEvent>(HandleCollide);
+        SubscribeLocalEvent<TurnstileComponent, PreventCollideEvent>(HandlePreventCollide);
+    }
+
+    private void HandlePreventCollide(Entity<TurnstileComponent> ent, ref PreventCollideEvent args)
+    {
+        // If Turnstile is rotating, allow entities the admitted entity is pulling to come through with it.
+        if (ent.Comp.State != TurnstileState.Rotating || !TryComp<PreventCollideComponent>(ent, out var preventCollide))
+            return;
+
+        if (TryComp<SharedPullerComponent>(preventCollide.Uid, out var puller))
+            args.Cancelled |= puller.Pulling == args.OtherEntity;
+    }
+
+    private void OnComponentInit(Entity<TurnstileComponent> ent, ref ComponentInit args)
+    {
+        // No turnstile should be spawned in the rotating state, as that requires a mob to be passing through.
+        if (ent.Comp.State == TurnstileState.Rotating)
+            ent.Comp.State = TurnstileState.Idle;
```

Comment:
This going to cause save/load problems because you're saving it but then immediately bulldozing it anyway.

Reply:
State removed from component - now the only marker of state is whether or not the turnstile has its ActiveTurnstileMarkerComponent attached, which will expire & be removed when nothing intersects the turnstile.